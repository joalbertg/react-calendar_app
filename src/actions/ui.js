import types from '~types';

export const uiOpenModal = () => ({
  type: types.UI_OPEN_MODAL
});

export const uiCloseModal = () => ({
  type:types.UI_CLOSE_MODAL
});

export const setError = err => ({
  type: types.UI_SET_ERROR,
  payload: err
});

export const removeError = () => ({
  type: types.UI_REMOVE_ERROR
});

