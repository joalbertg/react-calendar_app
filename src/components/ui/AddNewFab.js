import React from 'react';
import { useDispatch } from 'react-redux';

import { uiOpenModal, eventClearActiveEvent } from '~actions';

export const AddNewFab = () => {
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(eventClearActiveEvent());
    dispatch(uiOpenModal());
  }

  return(
    <button
      onClick={handleClick}
      className='btn btn-primary fab'
    >
      <i className='fas fa-plus'></i>
    </button>
  );
}

