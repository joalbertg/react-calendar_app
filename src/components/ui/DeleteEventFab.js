import React from 'react';
import { useDispatch } from 'react-redux';

import { eventStartDelete } from '~actions';

export const DeleteEventFab = () => {
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(eventStartDelete());
  }

  return(
    <button
      onClick={handleClick}
      className='btn btn-danger fab-danger'
    >
      <i className='fas fa-trash'></i>
    </button>
  );
}

