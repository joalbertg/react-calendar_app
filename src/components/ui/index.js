export { Navbar } from './Navbar';
export { AddNewFab } from './AddNewFab';

