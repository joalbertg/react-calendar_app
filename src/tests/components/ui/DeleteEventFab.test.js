import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';

import { DeleteEventFab } from '~components/ui';
import { eventStartDelete } from '~actions';

jest.mock('~actions', () => ({
  eventStartDelete: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);

store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
);

describe('Test DeleteEventFab component', () => {
  test('Should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should call to dispatch', async () => {
    wrapper.find('button').prop('onClick')();

    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(eventStartDelete).toHaveBeenCalled();
  });
});

