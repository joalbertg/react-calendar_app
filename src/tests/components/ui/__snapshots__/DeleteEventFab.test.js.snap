// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DeleteEventFab component Should display correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
</button>
`;
